<?xml version="1.0"?>

<sdf version='1.6'>
	<model name="robotic_arm">
		
		<static>false</static>
		<!-- Ombro -->
		<link name="ombro_link">
			<gravity> 0 </gravity>
			<pose> 0 0 1 0 0 0 </pose>
			<frame name="ombro_frame"/>
			
			<visual name="ombro_visual">
				<geometry>
					<box>
						<size> 0.1 0.05 0.05 </size>
					</box>
				</geometry>
				<material>
					<ambient>0 0.4 0 1</ambient>
					<diffuse>0 0.4 0 1</diffuse>
					<specular>0 0 0 1</specular>
					<emissive>0 0 0 0</emissive>
				</material>
			</visual>
			<collision name="ombro_collision">
				<geometry>
					<box>
						<size> 0.1 0.05 0.05 </size>
					</box>
				</geometry>
			</collision>
			<inertial>
				<mass> 0.3 </mass>
				<inertia>
					
					<ixx> 0.000125 </ixx>
					<iyy> 0.00031250000000000006 </iyy>
					<izz> 0.00031250000000000006 </izz>
					<ixy>0.0</ixy>
					<ixz>0.0</ixz>
					<iyz>0.0</iyz>
				</inertia>
			</inertial>
		</link>
		
		<!-- Box0 -->
		<link name="Box0_link">
			<gravity> 0 </gravity>
			
			<pose> 0 0.08 1 0 0 0 </pose>
			<visual name="Box0_visual">
				<geometry>
					<box>
						<size> 0.06 0.04 0.03 </size>
					</box>
				</geometry>
				<material>
					<ambient>0 0 0 1</ambient>
					<diffuse>0 0 0 1</diffuse>
					<specular>0 0 0 1</specular>
					<emissive>0 0 0 0</emissive>
				</material>
			</visual>
			<collision name="Box0_collision">
				<geometry>
					<box>
						<size> 0.06 0.04 0.03 </size>
					</box>
				</geometry>
			</collision>
			<inertial>
				<mass> 0.3 </mass>
				<inertia>
					
					<ixx> 6.25e-05 </ixx>
					<iyy> 0.00011249999999999998 </iyy>
					<izz> 0.00013 </izz>
					<ixy> 0.0 </ixy>
					<ixz> 0.0 </ixz>
					<iyz> 0.0 </iyz>
				</inertia>
			</inertial>
		</link>
		
		<!-- UMERO -->
		<link name="Tube0_link">
			<gravity> 0 </gravity>
			
			<pose> 0 0.1 1 0 0 0 </pose>
			<visual name="Tube0_visual">
				<pose> 0 0.1 0 1.57075 0 0 </pose>
				<geometry>
					<cylinder>
						<length> 0.2 </length>
						<radius> 0.01 </radius>
					</cylinder>
				</geometry>
				<material>
					<ambient>0.6 0.6 0.6 1</ambient>
					<diffuse>0.6 0.6 0.6 1</diffuse>
					<specular>0 0 0 1</specular>
					<emissive>0 0 0 0</emissive>
				</material>
			</visual>
			<collision name="Tube0_collision">
				<pose> 0 0.1 0 1.57075 0 0 </pose>
				<geometry>
					<cylinder>
						<length> 0.2 </length>
						<radius> 0.01 </radius>
					</cylinder>
				</geometry>
			</collision>
			<inertial>
				<pose> 0 0.1 0 1.57075 0 0 </pose>
				<mass> 0.1 </mass>
				<inertia>
					
					<ixx> 0.00033583333333333344 </ixx>
					<iyy> 0.00033583333333333344 </iyy>
					<izz> 5.0e-06 </izz>
					<ixy> 0.0 </ixy>
					<ixz> 0.0 </ixz>
					<iyz> 0.0 </iyz>
				</inertia>
			</inertial>
		</link>
		
		<!-- Box1 -->
		<link name="Box1_link">
			<gravity> 0 </gravity>
			
			<pose> 0 0.32500000000000007 1 0 0 0 </pose>
			<visual name="Box1_visual">
				<geometry>
					<box>
						<size> 0.03 0.05 0.02 </size>
					</box>
				</geometry>
				<material>
					<ambient>0 0 0 1</ambient>
					<diffuse>0 0 0 1</diffuse>
					<specular>0 0 0 1</specular>
					<emissive>0 0 0 0</emissive>
				</material>
			</visual>
			<collision name="Box1_collision">
				<geometry>
					<box>
						<size> 0.03 0.05 0.02 </size>
					</box>
				</geometry>
			</collision>
			<inertial>
				<mass> 0.3 </mass>
				<inertia>
					
					<ixx> 7.250000000000001e-05 </ixx>
					<iyy> 3.25e-05 </iyy>
					<izz> 8.499999999999999e-05 </izz>
					<ixy> 0.0 </ixy>
					<ixz> 0.0 </ixz>
					<iyz> 0.0 </iyz>
				</inertia>
			</inertial>
		</link>
		
		<!-- Box2 -->
		<link name="Box2_link">
			<gravity> 0 </gravity>
			
			<pose> 0 0.38500000000000006 1 0 0 0 </pose>
			<visual name="Box2_visual">
				<geometry>
					<box>
						<size> 0.05 0.03 0.02 </size>
					</box>
				</geometry>
				<material>
					<ambient>0 0 0 1</ambient>
					<diffuse>0 0 0 1</diffuse>
					<specular>0 0 0 1</specular>
					<emissive>0 0 0 0</emissive>
				</material>
			</visual>
			<collision name="Box2_collision">
				<geometry>
					<box>
						<size> 0.05 0.03 0.02 </size>
					</box>
				</geometry>
			</collision>
			<inertial>
				<mass> 0.3 </mass>
				<inertia>
					
					<ixx> 3.25e-05 </ixx>
					<iyy> 7.250000000000001e-05 </iyy>
					<izz> 8.499999999999999e-05 </izz>
					<ixy> 0.0 </ixy>
					<ixz> 0.0 </ixz>
					<iyz> 0.0 </iyz>
				</inertia>
			</inertial>
		</link>
		
		<!-- Tube1 -->
		<link name="Tube1_link">
			<gravity> 0 </gravity>
			
			<pose> 0 0.4 1 0 0 0 </pose>
			<visual name="Tube1_visual">
				<pose> 0 0.1 0 1.57075 0 0 </pose>
				<geometry>
					<cylinder>
						<length> 0.2 </length>
						<radius> 0.01 </radius>
					</cylinder>
				</geometry>
				<material>
					<ambient>0.6 0.6 0.6 1</ambient>
					<diffuse>0.6 0.6 0.6 1</diffuse>
					<specular>0 0 0 1</specular>
					<emissive>0 0 0 0</emissive>
				</material>
			</visual>
			<collision name="Tube1_collision">
				<pose> 0 0.1 0 1.57075 0 0 </pose>
				<geometry>
					<cylinder>
						<length> 0.2 </length>
						<radius> 0.01 </radius>
					</cylinder>
				</geometry>
			</collision>
			<inertial>
				<pose> 0 0.1 0 1.57075 0 0 </pose>
				<mass> 0.1 </mass>
				<inertia>
					
					<ixx> 0.00033583333333333344 </ixx>
					<iyy> 0.00033583333333333344 </iyy>
					<izz> 5.0e-06 </izz>
					<ixy> 0.0 </ixy>
					<ixz> 0.0 </ixz>
					<iyz> 0.0 </iyz>
				</inertia>
			</inertial>
		</link>
		
		<!-- Box3 -->
		<link name="Box3_link">
			<gravity> 0 </gravity>
			
			<pose> 0 0.6250000000000001 1 0 0 0 </pose>
			<visual name="Box3_visual">
				<geometry>
					<box>
						<size> 0.03 0.05 0.02 </size>
					</box>
				</geometry>
				<material>
					<ambient>0 0 0 1</ambient>
					<diffuse>0 0 0 1</diffuse>
					<specular>0 0 0 1</specular>
					<emissive>0 0 0 0</emissive>
				</material>
			</visual>
			<collision name="Box3_collision">
				<geometry>
					<box>
						<size> 0.03 0.05 0.02 </size>
					</box>
				</geometry>
			</collision>
			<inertial>
				<mass> 0.3 </mass>
				<inertia>
					
					<ixx> 7.250000000000001e-05 </ixx>
					<iyy> 3.25e-05 </iyy>
					<izz> 8.499999999999999e-05 </izz>
					<ixy> 0.0 </ixy>
					<ixz> 0.0 </ixz>
					<iyz> 0.0 </iyz>
				</inertia>
			</inertial>
		</link>
		
		<!-- Gripper -->
		<link name="Gripper_link">
			<gravity> 0 </gravity>
			
			<pose> 0 0.7200000000000002 1 0 0 0 </pose>
			<visual name="Gripper_visual">
				<geometry>
					<box>
						<size> 0.1 0.1 0.03 </size>
					</box>
				</geometry>
				<material>
					<ambient>0 0 0.5 1</ambient>
					<diffuse>0 0 0.5 1</diffuse>
					<specular>0 0 0 1</specular>
					<emissive>0 0 0 0</emissive>
				</material>
			</visual>
			<collision name="Gripper_collision">
				<geometry>
					<box>
						<size> 0.1 0.1 0.03 </size>
					</box>
				</geometry>
			</collision>
			<inertial>
				<mass> 0.25 </mass>
				<inertia>
					
					<ixx> 0.00022708333333333337 </ixx>
					<iyy> 0.00022708333333333337 </iyy>
					<izz> 0.00041666666666666675 </izz>
					<ixy> 0.0 </ixy>
					<ixz> 0.0 </ixz>
					<iyz> 0.0 </iyz>
				</inertia>
			</inertial>
		</link>

		<!-- Joints -->
		<joint name="joint0" type="fixed">
			<gravity>  </gravity>
			<parent>world</parent>
			<child>ombro_link</child>
			<pose> 0 0 1 0 0 0 </pose>
		</joint>
		
		<joint name="joint1" type="revolute">
			<parent>ombro_link</parent>
			<child>Box0_link</child>
			<pose> 0 -0.08 0 0 0 0 </pose>
			<axis>
				<xyz>1 0 0</xyz>
				<limit>
					<lower>-1.57075</lower>
					<upper>1.57075</upper>
					<effort>10000</effort>
					<velocity>0.01</velocity>
				</limit>
			</axis>
		</joint>
		
		<joint name="joint2" type="revolute">
			<gravity>  </gravity>
			<parent>Box0_link</parent>
			<child>Tube0_link</child>
			<pose> 0 0 0 0 0 0 </pose>
			<axis>
				<xyz>0 1 0</xyz>
				<limit>
					<lower>-3.1416</lower>
					<upper>3.1416</upper>
					<effort>100</effort>
					<velocity>0.01</velocity>
				</limit>
			</axis>
		</joint>
		
		<joint name="joint3" type="fixed">
			<parent>Tube0_link</parent>
			<child>Box1_link</child>
			<pose> 0 0 0 0 0 0 </pose>
		</joint>
		
		<joint name="joint4" type="revolute">
			<parent>Box1_link</parent>
			<child>Box2_link</child>
			<pose> 0 -0.035 0 0 0 0 </pose>
			<axis>
				<xyz>1 0 0</xyz>
				<limit>
					<lower>-3</lower>
					<upper>3</upper>
					<effort>100</effort>
					<velocity>0.01</velocity>
				</limit>
			</axis>
		</joint>
		
		<joint name="joint5" type="revolute">
			<parent>Box2_link</parent>
			<child>Tube1_link</child>
			<pose> 0 0 0 0 0 0 </pose>
			<axis>
				<xyz>0 1 0</xyz>
				<limit>
					<lower>-3.1416</lower>
					<upper>3.1416</upper>
					<effort>100</effort>
					<velocity>0.01</velocity>
				</limit>
			</axis>
		</joint>
		
		<joint name="joint6" type="fixed">
			<parent>Tube1_link</parent>
			<child>Box3_link</child>
			<pose> 0 0 0 0 0 0 </pose>
		</joint>
		
		<joint name="joint7" type="revolute">
			<parent>Box3_link</parent>
			<child>Gripper_link</child>
			<pose> 0 -0.08 0 0 0 0 </pose>
			<axis>
				<xyz>1 0 0</xyz>
				<limit>
					<lower>-2</lower>
					<upper>2</upper>
					<effort>100</effort>
					<velocity>0.01</velocity>
				</limit>
			</axis>
		</joint>
		
	</model>
</sdf>
